hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -105,7 +105,8 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
 
     return getOmRequest().toBuilder()
         .setCommitKeyRequest(commitKeyRequest.toBuilder()
-            .setKeyArgs(newKeyArgs)).setUserInfo(getUserInfo()).build();
+            .setKeyArgs(newKeyArgs))
+        .setUserInfo(getUserIfNotExists(ozoneManager)).build();
```

Comment:
The same `setUserInfo(getUserIfNotExists` logic is performed in `OMClientRequest#preExecute` (the base method in parent class).

https://github.com/apache/ozone/blob/40e6875a7e6b5be1d6bf617a441e1c9aab1eb72b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java#L98-L107

I wonder if `OMKeyCommitRequest` should call the parent method first thing, like some other requests:

https://github.com/apache/ozone/blob/40e6875a7e6b5be1d6bf617a441e1c9aab1eb72b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/security/OMGetDelegationTokenRequest.java#L64-L67

Reply:
@adoroszlai Calling parent method will not work here as building of request is different, parent does same thing with dummy request building.

Reply:
@sumitagrawl can you please explain what you mean by "dummy request building"?

Reply:
@adoroszlai Its not dummy, its original request building with setting only userInfo and layout.
To support modification of request content to build by parent, a builder need to be exposed in parent class, which derived class can manipulate.

But still we are having 2 way to get UserInfo() ---

OMClientRequest {
    OMRequest.Builder builder = getOmRequest().toBuilder();
    public void preExecute(...) {
      builder
      .setUserInfo(getUserIfNotExists(ozoneManager))
      .build();
   }
}
DeriveClass extends OMClientRequest {
    public void preExecute(...) {
      // do modification
     builder.setCommitKeyRequest(...);
   }
}

Reply:
@sumitagrawl `OMGetDelegationTokenRequest` and similar requests call `super.preExecute` at the start of the method to obtain the `OMRequest` updated by parent logic, then update with local changes as necessary.  It is not optimal, due to duplicate builder conversion, but it is correct.

Reply:
@adoroszlai Thanks for valuable input, I have updated the changes